### PR TITLE
Add POST persons/create endpoint

### PR DESCRIPTION
**Pull request by sydive**
* Added the `http-status` library to the backend to make sending responses easier
* Reconfiguring of routes/port number
    * `src/index.ts` now contains the route of `/api` instead of `src/routes/index.ts`
    * `src/routes/index.ts` now contains the route `/persons`
    * `src/routes/person.route.ts` now contains the route `/create`
    * The port number used in `src/index.ts` was changed to 5000 from 3000, in order to stop conflicting with the frontend (which usually also tries to default to 3000)
* Implemented the POST `persons/create` endpoint
* Added a new TypeScript type to use for POST requests

### CODE REVIEW

File: backend/src/controllers/person.controller.ts [hunk 1]
```diff
@@ -1,31 +1,21 @@
...
-  const { name, organisation } = req.body;
   try {
-    await createPersonService({ name, organisation });
-    res.status(200).end();
+    const person = await personService.createPerson(req.body);
+    res.status(httpStatus.CREATED).send(person);
   } catch (e) {
-    next(e);
+    logger.error(e);
+    res.status(httpStatus.BAD_REQUEST);
```
Comment (hajineats) [inline backend/src/controllers/person.controller.ts]: Was there a particular reason you avoided using next(e)? My rationale behind using next(e) is that you can then pass error handling to the errorHandler defined in `backend/src/utils/middleware`. This way, you don't need to do logger.error(e) and res.status() every time you handle error

File: backend/src/index.ts [hunk 2]
```diff
@@ -11,12 +11,12 @@ connect(config.MONGODB_URI)
   .catch((err) => logger.error('error connecting to mongodb:', err.message));
 
 const app = express();
-const port = 3000;
+const port = 5000;
```
Comment (hajineats) [inline backend/src/index.ts]: I forgot to do this but we can probably define this in .env and reference it from config.js

File: backend/src/index.ts [hunk 3]
```diff
@@ -11,12 +11,12 @@ connect(config.MONGODB_URI)
...
 
 app.use(express.urlencoded({ extended: false }));
 app.use(cors());
 app.use(express.json());
-app.use('/', routes);
+app.use('/api', routes);
```
Comment (hajineats) [inline backend/src/index.ts:18]: good idea

File: backend/src/controllers/person.controller.ts [hunk 4]
```diff
@@ -1,31 +1,21 @@
...
-  req: Request,
-  res: Response,
-  next: NextFunction,
-): Promise<void> => {
-  const { name, organisation } = req.body;
   try {
-    await createPersonService({ name, organisation });
-    res.status(200).end();
+    const person = await personService.createPerson(req.body);
+    res.status(httpStatus.CREATED).send(person);
```
Comment (hajineats) [inline backend/src/controllers/person.controller.ts:20]: Yep I like this change

File: backend/src/controllers/controller.types.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { Request, Response, ParamsDictionary, Paths } from 'express';
+
+export type POST = (
+    req: Request<ParamsDictionary, {}, Paths.Hangouts.Post.RequestBody>,
+    res: Response<
+        | Paths.Hangouts.Post.Responses.$201
+        | Paths.Hangouts.Post.Responses.$400
+        | Paths.Hangouts.Post.Responses.$404
+    >
+) => void
```
Comment (hajineats) [inline backend/src/controllers/controller.types.ts]: is this meant to be `Promise<void>`?

**Review by hajineats — COMMENTED** [review]
Added some comments on the change

File: backend/src/index.ts (hunk 2 above)
Comment (hajineats) [reply backend/src/index.ts]: usually it's 3000 for FE and 3001 for BE but doesn't really matter once we have it as .env

File: backend/src/controllers/controller.types.ts (hunk 5 above)
Comment (sydive) [reply backend/src/controllers/controller.types.ts]: Yep it should be

File: backend/src/controllers/person.controller.ts (hunk 1 above)
Comment (sydive) [reply backend/src/controllers/person.controller.ts]: I actually completely missed this, I think if I just replace the current `400` value in the errorHandler with `httpStatus.BAD_REQUEST` it should work the same 👍

File: backend/src/index.ts (hunk 2 above)
Comment (sydive) [reply backend/src/index.ts]: True I can swap the values out now

**Comment by hajineats** [comment]
looks good to me! raise PR to 701 repo whenever you are ready :)

**Pull request closed without merging**